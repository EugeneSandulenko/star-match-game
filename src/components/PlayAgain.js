import React from 'react';

class PlayAgain extends React.Component {
  state ={
    name : 'Eugene',
  };

  render() {
    return (
      <div className="game-done">
        <div
          className="message"
          style={{ color: this.props.gameStatus === 'lost' ? 'red' : 'green' }}
        >
          {this.props.gameStatus === 'lost' ? 'Game Over' : 'Nice'}
        </div>
        <label>
          Name
          <input
            type="text"
            name="name"
            placeholder="Enter your name"
            value={this.state.name}
            onChange={event => this.setState({ name: event.target.value })}
          />
        </label>
        <br />
        <button onClick={()=>  this.props.onClick(this.state.name)}>Play Again</button>
      </div>
    );
  }
}

export default PlayAgain;
