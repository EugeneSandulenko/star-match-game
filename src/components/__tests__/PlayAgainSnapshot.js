import React from 'react';
import PlayAgain from '../PlayAgain';
import renderer from 'react-test-renderer';

describe('PlayAgain', () => {
  it('renders PlayAgain correctly', () => {
    const tree = renderer
      .create(<PlayAgain onClick={() => {}} gameStatus="Active" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
