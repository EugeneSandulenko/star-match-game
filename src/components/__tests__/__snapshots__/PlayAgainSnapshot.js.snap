// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayAgain renders PlayAgain correctly 1`] = `
<div
  className="game-done"
>
  <div
    className="message"
    style={
      Object {
        "color": "green",
      }
    }
  >
    Nice
  </div>
  <label>
    Name
    <input
      name="name"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
  </label>
  <br />
  <button
    onClick={[Function]}
  >
    Play Again
  </button>
</div>
`;
